File: src/main/java/com/cpjit/swagger4j/Operation.java
/*
 * Copyright 2011-2019 CPJIT Group.
 *
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package com.cpjit.swagger4j;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

/**
 * 该类原名叫Path，在2.2.0开始被重命名为Operation。
 *
 * @author yonghaun
 * @since 1.0.0
 */
class Operation {
    private String summary;
    private String description;
    private String operationId;
    private List<String> tags = new ArrayList<>();
    private List<String> consumes = new ArrayList<>();
    private List<String> produces = new ArrayList<>();
    private List<Map<String, Object>> parameters = new ArrayList<>();
    /**
     * @since 1.2.2
     */
    private boolean deprecated;
    /**
     * @since 2.2.0。
     */
    private Map<String, Object> responses;

    public String getSummary() {
        return summary;
    }

    public void setSummary(String summary) {
        this.summary = summary;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getOperationId() {
        return operationId;
    }

    public void setOperationId(String operationId) {
        this.operationId = operationId;
    }

    public List<String> getTags() {
        return tags;
    }

    public void setTags(List<String> tags) {
        this.tags = tags;
    }

    public List<String> getConsumes() {
        return consumes;
    }

    public void setConsumes(List<String> consumes) {
        this.consumes = consumes;
    }

    public List<String> getProduces() {
        return produces;
    }

    public void setProduces(List<String> produces) {
        this.produces = produces;
    }

    public List<Map<String, Object>> getParameters() {
        return parameters;
    }

    public void setParameters(List<Map<String, Object>> parameters) {
        this.parameters = parameters;
    }

    /**
     * 判断接口是否已经被废弃。
     */
    public boolean isDeprecated() {
        return deprecated;
    }

    public void setDeprecated(boolean deprecated) {
        this.deprecated = deprecated;
    }

    public Map<String, Object> getResponses() {
        return responses;
    }

    public Operation setResponses(Map<String, Object> responses) {
        this.responses = responses;
        return this;
    }
}

File: src/main/kotlin/com/cpjit/swagger4j/specification/Operation.kt
/*
 * Copyright 2011-2018 CPJIT Group.
 *
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package com.cpjit.swagger4j.specification


/**
 * @author yonghuan
 * @since 2.1.6
 */
class Operation(val operationId: String) {

    var tags: Array<String>? = null
    var summary: String? = null
    var description: String? = null
    var consumes: Array<String>? = null
    var produces: Array<String>? = null
    /**
     * The transfer protocol for the operation. Values MUST be from the list: "http", "https", "ws", "wss".
     */
    var schemes: Array<String>? = null
    var deprecated = false
    var parameters: List<Parameter>? = null
}